### PR TITLE
[Java][JAXRS-CXF] Improve API documentation in the CXF Server stub and Client generation

### LINKED ISSUE #6696: [Java][CXF] Include more of the API documentation in the CXF Server stub generation
##### Description

This is a feature request. I want to include more of the API documentation in the CXF Server stub generation so that the Java code is more complete.

##### Swagger-codegen version

2.3.0

##### Swagger declaration file content or url
..

##### Command line used for generation

..

##### Steps to reproduce

..

##### Related issues/PRs

..

##### Suggest a fix/enhancement

Here is the change that I propose: https://github.com/gustavoapaz/swagger-codegen/commit/05eefe2adbde8a9b8cb44c7bdccd5a81ab9daba3

### PR DESCRIPTION
**Pull request by gustavoapaz**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [x] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming langauge.

### Description of the PR

I included more of the API documentation in the CXF Server stub and Client generation. 

fix #6696

@bbdouglas, @JFCote, @sreeshas, @jfiala, @lukoyanov, @cbornet, @wing328

### CODE REVIEW

**Comment by JFCote** [comment]
I'm not sure if this should be by default or by adding a CLI options. @wing328, what do you think?

In the Play Framework, I don't put any documentation in the generated code since the documentation is the spec itself (YAML). It's like duplicating information at 2 places. But I guess there are scenario when this is needed so a CLI options might be best.

**Comment by gustavoapaz** [comment]
@ricardona what do you think about this?

**Comment by wing328** [comment]
> In the Play Framework, I don't put any documentation in the generated code since the documentation is the spec itself (YAML). 

In some other server generators, I think we do include the doc in the auto-generated code and the IDE should be able to display those when the developers are working on the code.

In API clients, we definitely put the doc (summary, notes, description) in the code as the one using the API clients may not have the spec with them.

**Comment by JFCote** [comment]
@wing328 Yes, for the client it makes total sense but for the server, not so much. But if other server do it, I won't object.

File: modules/swagger-codegen/src/main/resources/JavaJaxRS/cxf/pom.mustache [hunk 1]
```diff
@@ -4,6 +4,9 @@
   <artifactId>{{artifactId}}</artifactId>
   <packaging>jar</packaging>
   <name>{{artifactId}}</name>
+  {{#appDescription}}
+  <description>{{{appDescription}}}</description>
```
Comment (wing328) [inline modules/swagger-codegen/src/main/resources/JavaJaxRS/cxf/pom.mustache]: @gustavoapaz I think you will need to use {{appDescription}} instead to use the HTML-escaped value in handling special characters like <, > in the description.

**Pull request merged**